### PR TITLE
Algolia-mesh: add category search

### PR DESCRIPTION
**Pull request by Renzovh**

### CODE REVIEW

File: packages/algolia-mesh/mesh/categories/getCategoryResults.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { MeshContext, QueryproductsArgs } from '@graphcommerce/graphql-mesh'
+import type { GraphQLResolveInfo } from 'graphql'
+import { getIndexName } from '../getIndexName'
+import { getCategoryResultsInput } from './getCategoryResultsInput'
+
+export async function getCategoryResults(
+  args: QueryproductsArgs,
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/getCategoryResults.ts:7]: It should never bet Queryproductargs. I would expect something related to categories.

File: packages/algolia-mesh/mesh/categories/getCategoryResultsInput.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Queryalgolia_searchSingleIndexArgs, QueryproductsArgs } from '@graphcommerce/graphql-mesh'
+
+// eslint-disable-next-line @typescript-eslint/require-await
+export async function getCategoryResultsInput(
+  args: QueryproductsArgs,
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/getCategoryResultsInput.ts:5]: Querycategoriesargs?

File: packages/algolia-mesh/mesh/categories/getCategoryResults.ts (hunk 1 above)
Comment (paales) [reply packages/algolia-mesh/mesh/categories/getCategoryResults.ts:7]:
```
filters: {name: { match: "men" }}
```

File: packages/algolia-mesh/mesh/categories/resolvers.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { type Resolvers } from '@graphcommerce/graphql-mesh'
+import { getStoreConfig } from '../getStoreConfig'
+import { algoliaHitToMagentoCategory, CategoriesItemsItem } from './algoliaHitToMagentoCategory'
+import { getCategoryResults } from './getCategoryResults'
+
+export const resolvers: Resolvers = {
+  Query: {
+    categories: async (root, args, context, info) => {
+      const isAgolia =
+        (args.filters?.categories_engine?.in ?? [args.filters?.categories_engine?.eq])[0] ===
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/resolvers.ts:10]: engine

File: packages/algolia-mesh/mesh/categories/resolvers.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        'algolia'
+
+      if (!isAgolia) return context.m2.Query.categories({ root, args, context, info })
+      const algoliaResponse = await getCategoryResults(args, context, info)
+      const items: (CategoriesItemsItem | null)[] = []
+      const storeConfig = await getStoreConfig(context)
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/resolvers.ts:16]: Promise.all

File: packages/algolia-mesh/mesh/categories/resolvers.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          items.push(category)
+        }
+      }
+
+      return {
+        items,
+        page_info: {
+          current_page: 1,
+          page_size: 20,
+          total_pages: 1,
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/resolvers.ts:40]: Todo

File: packages/algolia-mesh/mesh/categories/resolvers.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        },
+        total_count: 0,
```
Comment (paales) [inline packages/algolia-mesh/mesh/categories/resolvers.ts:42]: todo

File: packages/algolia-mesh/mesh/getIndexName.ts [hunk 7]
```diff
@@ -5,7 +5,7 @@ function getStoreHeader(context: MeshContext) {
   return (context as MeshContext & { headers: Record<string, string | undefined> }).headers.store
 }
 
-export function getIndexName(context: MeshContext) {
+export function getIndexName(context: MeshContext, indexType: string = 'products') {
```
Comment (paales) [inline packages/algolia-mesh/mesh/getIndexName.ts:8]: Create a getIndexName inside categories/getIndexName.ts

File: packages/algolia-mesh/mesh/getSearchSuggestionsInput.ts [hunk 8]
```diff
@@ -18,5 +19,6 @@ export async function getSearchSuggestionsInput(
     query: search,
     hitsPerPage: 5,
     analytics: true,
+    userToken: getUserToken(context),
```
Comment (paales) [inline packages/algolia-mesh/mesh/getSearchSuggestionsInput.ts:22]: Please check the `personalizationEnabled` config. Do we want to keep that? If so, shouldn't this be moved to another plugin (OR should we check the config value here?)

File: packages/algolia-mesh/plugins/magentoSearchApplyAlgoliaCategoriesEngine.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+// export const productListApplySearchDefaults: FunctionPlugin<typeof productListApplyDefaults> = (
+//   prev,
+//   params,
+//   conf,
+// ) => applyEngineVariables(prev(params, conf))
+
+// export const searchDefaultsToProductListFilters: FunctionPlugin<
+//   typeof defaultsToProductListFilters
+// > = (prev, variables) => applyEngineVariables(prev(variables))
```
Comment (paales) [inline packages/algolia-mesh/plugins/magentoSearchApplyAlgoliaCategoriesEngine.ts:26]: Empty file, remove?

**Review by paales — CHANGES_REQUESTED** [review]

File: packages/algolia-mesh/mesh/getSearchSuggestionsInput.ts (hunk 8 above)
Comment (Renzovh) [reply packages/algolia-mesh/mesh/getSearchSuggestionsInput.ts:22]: I deleted the personalization config, while indeed this is managed in the algolia dashboard

**Pull request merged**